primitives/trie/src/lib.rs

```diff
@@ -856,4 +852,34 @@ mod tests {
 			).is_err()
 		);
 	}
+
```

Comment:
I am not totally sure if we really need this test, but having test is better.

Reply:
We need this test, clearly! I don't want to spend another two weeks chasing the same bug again :P